Cleanup pass in setup.py

romanc opened the pull request:
**Description**

This PR does a pass on the `setup.py` file and changes the following

- Fully move ownership information from AI2 to NOAA-GDFL
- Restrict python version to 3.11 ([NDSL requirement](https://github.com/NOAA-GFDL/NDSL?tab=readme-ov-file#requirements--supported-compilers))
- Consistent license information
- Pin NDSL at latest release, 2024.09.00, see https://github.com/NOAA-GFDL/PyFV3/pull/23#discussion_r1668732097

This is a follow-up from  a couple of PRs and mentions including

- https://github.com/NOAA-GFDL/PyFV3/pull/21#discussion_r1671043186 and https://github.com/NOAA-GFDL/PyFV3/pull/21#pullrequestreview-2175496120 which advocated for a holistic pass on the file
- https://github.com/NOAA-GFDL/PyFV3/pull/27 which only updated python version

**How Has This Been Tested?**

Tested `python_requires` locally (with different python version and multiple virtual environments)

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [ ] New check tests, if applicable, are included
- [ ] Targeted model if this changed was triggered by a model need/shortcoming

Commit:
Cleanup pass in setup.py

- Fully move from AI2 to NOAA-GDFL
- Restrict python version to 3.11
- Consistent license information
- Pin NDSL at latest release (2024.09.00)

romanc commented:
@fmalatino @oelbert would you mind to review this? And can someone else add Rusty to the list of reviewers? I can't seem to find him.

I'd add you guys as reviewers if I could. Looks like I don't have the rights to do this.

/cc @FlorianDeconinck as discussed.

oelbert reviewed (dismissed):
Looks good to me! Thanks for taking care of this.

bensonr commented on setup.py:
```diff
@@ -15,35 +13,46 @@
         "Development Status :: 2 - Pre-Alpha",
         "Intended Audience :: Developers",
-        "License :: OSI Approved :: BSD License",
+        "License :: OSI Approved :: GPLv3 License",
```
Are we going to use the GPL or LGPL at this point in time?

FlorianDeconinck replied:
GT4Py & DaCe are BSD - so they don't care.

It boils down to the default NOAA license. If you guys use LGPL by default we should go there probably.

FlorianDeconinck replied:
But then we should change the License in the repository - right now it's GPLv3

romanc replied:
At this point in time you license is GPLv3 (as stated in the `LICENSE.txt`). I don't have any plans to change the license. That meta information was just plain wrong.

In case you (as-in "the NOAA people, who are owning this repository") want to have a license discussion, feel free to do so. You, as the owner of the repo, should decide that. Maybe https://choosealicense.com/licenses/ or https://opensource.org/licenses can help with that decision.

I just want consistent information and currently the license GPLv3. Let me know what you decide. I'm happy to keep this PR open for some time and include a potential license change if that is what you want.

bensonr replied:
This must be the original license ascribed by AI2 as we generally use LGPLv3.  Let's leave it for now as Florian and I need to close the discussion loop with our higher ups to ensure we use a license complementary to each teams needs.

romanc replied:
As far as I can see, the original AI2 license is GPLv3, see [here](https://github.com/ai2cm/pace/blob/main/fv3core/LICENSE.txt).

A word of caution (I am really not a lawyer) about relicensing: LGPLv3 is less restrictive than GPLv3. You might want to seek legal advice on how to do this properly (from what I read [here](https://opensource.stackexchange.com/questions/33/how-can-a-project-be-relicensed)).

bensonr replied:
I will have to check with OliF to be sure, but I do recall a conversation about the handoff and when I inquired about licensing, I thought it was transferred to us (NOAA) free and clear.

romanc replied:
All good then. I'll close this conversation since it looks like we aren't planning on changing the license now. If we ever get there, we can adjust this meta information again.

bensonr commented on setup.py:
```diff
@@ -15,35 +13,46 @@
     long_description=readme,
     include_package_data=True,
-    name="pyFV3",
+    name="PyFV3",
```
We changed the `p` to uppercase here, but in the next line it is still lowercase.  Do we need consistency here?

romanc replied:
Good catch, Rusty. I shouldn't have changed the package name. Will revert the name.

To explain what these two lines do:
- The package name is used when writing imports like `from pyFV3 import ...`. We should thus keep `"pyFV3"` as-is.
- The include list of `find_namespace_packages()` expects folder names/paths to be included. Since the folder is called `"pyFV3"` (with lowercase p), we shouldn't change that either.

Commit:
Revert name change

romanc commented on setup.py:
```diff
@@ -15,32 +13,43 @@
-    license="BSD license",
+    license="GPLv3 license",
     long_description=readme,
     include_package_data=True,
```
Actually, I have a question on that line now that I see it. Does anyone know what the intention was here? While the default in the newer `project.toml` files is `True`, the default in `setup.py` is (and remains for compatibility reasons) `False`.

As I understand [the docs](https://setuptools.pypa.io/en/latest/userguide/quickstart.html#including-data-files), for this to have an effect, we'd need to provide a `MANIFEST.in` file or configure the git plugin to automatically add data files to the package...

FlorianDeconinck replied:
I think a while back (at AI2) the idea was to ship some translate data with the package. This was swiftly abandoned.

romanc replied:
In that case, let me just delete that line to avoid any future confusion

Commit:
Remove unused config option `include_package_data`

This was added a while back (at AI2) when - for a brief amount of time -
the idea as to ship some translate test data with the package. This idea
was rejected and now the configuration option isn't needed anymore.

romanc commented:
Thanks for all the discussions. This is ready for a second round of reviews.

FlorianDeconinck reviewed (approved):
LGTM

bensonr reviewed (approved):
lgtm

romanc commented:
Someone with write access to the repo will need to merge this PR. I don't have the rights to do this.
